### ISSUE TITLE
useOldImageOnUrlChange/gaplessPlayback not work properly

### ISSUE THREAD

**Issue by Peng-Qian** [Baseflow/flutter_cached_network_image#505]
## 🐛 Bug Report

when useOldImageOnUrlChange/gaplessPlayback set true, it throws an exception on the first time to load the new image

![1](https://user-images.githubusercontent.com/37643086/100090490-1c625a80-2eb8-11eb-80b0-1b5a0ba4b17c.gif)

### Expected behavior
animate from the old image to the new image
### Reproduction steps
```
import 'package:cached_network_image/cached_network_image.dart';
import 'package:flutter/material.dart';
import 'package:flutter_cache_manager/flutter_cache_manager.dart';
import 'package:octo_image/octo_image.dart';

void main() => runApp(LabApp());

class LabApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Lab',
      theme: ThemeData(
        primarySwatch: Colors.blueGrey,
        backgroundColor: Colors.white,
      ),
      home: const HomeScreen(),
    );
  }
}

class HomeScreen extends StatefulWidget {
  const HomeScreen();

  @override
  _HomeScreenState createState() => _HomeScreenState();
}

class _HomeScreenState extends State<HomeScreen> {
  bool isFirstImage = false;

  void changePicture() {
    setState(() {
      isFirstImage = !isFirstImage;
    });
  }

  String get url => isFirstImage
      ? 'https://boatingnz.co.nz/wp-content/uploads/2019/11/boating-nz-image-1075-underway.jpg'
      : 'https://helpx.adobe.com/content/dam/help/en/stock/how-to/visual-reverse-image-search/jcr_content/main-pars/image/visual-reverse-image-search-v2_intro.jpg';

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: const Text('Home Screen'),
      ),
      body: SizedBox.expand(
        child: Column(
          mainAxisAlignment: MainAxisAlignment.center,
          children: [
            SizedBox(
              height: 200,
              width: 200,
              child: CachedNetworkImage(
                imageUrl: url,
                fit: BoxFit.cover,
                useOldImageOnUrlChange: true,
                errorWidget: (context, url, error) => Icon(Icons.error),
              ),
            ),
            RaisedButton(
              child: Text('Change Picture'),
              onPressed: changePicture,
            ),
          ],
        ),
      ),
    );
  }
}

```
### Configuration

**Version:** 2.3.3

**Platform:** 
- [x] :iphone: iOS
- [x] :robot: Android

**Comment by troy379** [comment]
@renefloor is this going to be fixed?

**Comment by renefloor** [comment]
@troy379 this should already be fixed in version 3.0.0, do you still have this problem?
